<!DOCTYPE html>
<html lang="en-us">
  <head>
    <meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="generator" content="Hugo 0.55.6" />


<title>Gene expression - microarray data still useful - Pietro&#39;s website</title>
<meta property="og:title" content="Gene expression - microarray data still useful - Pietro&#39;s website">



  








<link href='//cdn.bootcss.com/highlight.js/9.11.0/styles/github.min.css' rel='stylesheet' type='text/css' />



<link rel="stylesheet" href="/css/fonts.css" media="all">
<link rel="stylesheet" href="/css/main.css" media="all">

<link rel="stylesheet" href="/css/mainoverwrite.css?v=1">



  </head>
  <body>
    <div class="wrapper">
      <header class="header">
        <nav class="nav">
  <a href="/" class="nav-logo">
    <img src="/images/pietro3.png"
         width="500"
         height="500"
         alt="Logo">
  </a>

  <ul class="nav-links">
    
    <li><a href="/about/">About</a></li>
    
    <li><a href="https://github.com/pietrod">GitHub</a></li>
    
    <li><a href="https://twitter.com/Pietro_Delfino">Twitter</a></li>
    
  </ul>
</nav>

      </header>


<main class="content" role="main">

  <article class="article">
    
    <span class="article-duration">5 min read</span>
    

    <h1 class="article-title">Gene expression - microarray data still useful</h1>

    
    <span class="article-date">2019/07/18</span>
    

    <div class="article-content">
      
<script src="/rmarkdown-libs/htmlwidgets/htmlwidgets.js"></script>
<script src="/rmarkdown-libs/jquery/jquery.min.js"></script>
<link href="/rmarkdown-libs/datatables-css/datatables-crosstalk.css" rel="stylesheet" />
<script src="/rmarkdown-libs/datatables-binding/datatables.js"></script>
<link href="/rmarkdown-libs/dt-core/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="/rmarkdown-libs/dt-core/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="/rmarkdown-libs/dt-core/js/jquery.dataTables.min.js"></script>
<link href="/rmarkdown-libs/crosstalk/css/crosstalk.css" rel="stylesheet" />
<script src="/rmarkdown-libs/crosstalk/js/crosstalk.min.js"></script>


<p>I work on pancreatic cancer <strong>-omics</strong>. Data mining on published data is a real source of gold today when someone wants to find confirmation of an hypothesis on a more general scale than his own dataset.</p>
<p>The most common -omics datasets out there are transcriptomics one; both RNA-Seq and good ol’ microarray. Regarding RNA-Seq, the well known TCGA and ICGC are my starting point for any explorative data mining analysis.</p>
<p>Regarding microarray, few studies with clear data have been published in the past, and someone has already collected the most interesting ones (see <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5674743/">Birnbaum et al. 2017</a>, <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5975421/">Zhao et al. 2018</a>).</p>
<p>For the scopes of my research I have also explored some of these datasets. The majority of theme are stored in the <a href="https://www.ncbi.nlm.nih.gov/geo/">GEO database</a>.</p>
<p>Here I want to present a basic workflow to retrieve a microarray expression dataset from GEO and to perform basic explorative analysis in <strong><em>R</em></strong>. As an example I take <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE62165">GSE62165</a> dataset.</p>
<p>To download and access data I use the <em>Bioconductor</em> <code>GEOquery</code> package (<a href="https://bioconductor.org/packages/release/bioc/html/GEOquery.html" class="uri">https://bioconductor.org/packages/release/bioc/html/GEOquery.html</a>).</p>
<pre class="r"><code># Load some libraries
library(tidyverse)
library(reshape2)
library(ggsci)
library(GEOquery)

# Downlaod ExpressionSet
x &lt;- getGEO(&#39;GSE62165&#39;, destdir = &#39;your/favourite/path&#39;, getGPL = T)</code></pre>
<p>We have downloaded the <em>GSE series matrix</em> to file and the <em>GPL .soft annotation</em>.
Now we load them back.</p>
<pre class="r"><code>x &lt;- getGEO(filename = &#39;your/favourite/path/GSE62165_series_matrix.txt.gz&#39;)</code></pre>
<p>Then we can start exploring the data</p>
<pre class="r"><code># First we save the expression matrix in a different object
mat &lt;- exprs(x)

dim(mat)</code></pre>
<p>[1] 49386 131</p>
<pre class="r"><code>kable(mat[1:8, 1:5], caption = &#39;Few lines of the expression matrix&#39;)</code></pre>
<table>
<caption><span id="tab:unnamed-chunk-4">Table 1: </span>Few lines of the expression matrix</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">GSM1520738</th>
<th align="right">GSM1520739</th>
<th align="right">GSM1520740</th>
<th align="right">GSM1520741</th>
<th align="right">GSM1520742</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>11715100_at</td>
<td align="right">3.412714</td>
<td align="right">3.736906</td>
<td align="right">3.300559</td>
<td align="right">3.223035</td>
<td align="right">3.073504</td>
</tr>
<tr class="even">
<td>11715101_s_at</td>
<td align="right">4.345207</td>
<td align="right">3.747892</td>
<td align="right">4.470069</td>
<td align="right">3.994475</td>
<td align="right">4.366361</td>
</tr>
<tr class="odd">
<td>11715102_x_at</td>
<td align="right">3.054823</td>
<td align="right">3.076922</td>
<td align="right">2.897363</td>
<td align="right">3.175010</td>
<td align="right">2.695467</td>
</tr>
<tr class="even">
<td>11715103_x_at</td>
<td align="right">4.510326</td>
<td align="right">4.156103</td>
<td align="right">3.456344</td>
<td align="right">4.064235</td>
<td align="right">4.227728</td>
</tr>
<tr class="odd">
<td>11715104_s_at</td>
<td align="right">3.229649</td>
<td align="right">3.367871</td>
<td align="right">3.030241</td>
<td align="right">3.249097</td>
<td align="right">2.973494</td>
</tr>
<tr class="even">
<td>11715105_at</td>
<td align="right">4.554059</td>
<td align="right">3.486847</td>
<td align="right">3.294781</td>
<td align="right">3.137451</td>
<td align="right">2.961364</td>
</tr>
<tr class="odd">
<td>11715106_x_at</td>
<td align="right">3.103925</td>
<td align="right">3.377799</td>
<td align="right">3.467268</td>
<td align="right">3.585230</td>
<td align="right">3.940851</td>
</tr>
<tr class="even">
<td>11715107_s_at</td>
<td align="right">3.082911</td>
<td align="right">4.107503</td>
<td align="right">3.012626</td>
<td align="right">3.033756</td>
<td align="right">3.222768</td>
</tr>
</tbody>
</table>
<p>Then we explore the distrbution of samples’ expression values with a boxplot</p>
<pre class="r"><code>boxplot(mat, outline=F, main=&#39;GSE62165&#39;)</code></pre>
<p><img src="/post/2019-07-17-microarray-still-useful_files/figure-html/unnamed-chunk-5-1.svg" width="672" /></p>
<p>What about annotation data? They are stored in <code>fData</code> slot. We save them in a unique object and look for the <strong>Gene Symbol</strong> column, that we will use to replace row names (probes id).</p>
<pre class="r"><code>ann &lt;- fData(x)

colnames(ann)</code></pre>
<p>[1] “ID” “GeneChip Array”<br />
[3] “Species Scientific Name” “Annotation Date”<br />
[5] “Sequence Type” “Sequence Source”<br />
[7] “Transcript ID(Array Design)” “Target Description”<br />
[9] “Representative Public ID” “Archival UniGene Cluster”<br />
[11] “UniGene ID” “Genome Version”<br />
[13] “Alignments” “Gene Title”<br />
[15] “Gene Symbol” “Chromosomal Location”<br />
[17] “GB_LIST” “SPOT_ID”<br />
[19] “Unigene Cluster Type” “Ensembl”<br />
[21] “Entrez Gene” “SwissProt”<br />
[23] “EC” “OMIM”<br />
[25] “RefSeq Protein ID” “RefSeq Transcript ID”<br />
[27] “FlyBase” “AGI”<br />
[29] “WormBase” “MGI Name”<br />
[31] “RGD Name” “SGD accession number”<br />
[33] “Gene Ontology Biological Process” “Gene Ontology Cellular Component”
[35] “Gene Ontology Molecular Function” “Pathway”<br />
[37] “InterPro” “Trans Membrane”<br />
[39] “QTL” “Annotation Description”<br />
[41] “Annotation Transcript Cluster” “Transcript Assignments”<br />
[43] “Annotation Notes”</p>
<pre class="r"><code># Subset to columns we&#39;ll use
ann &lt;- ann[,c(&#39;ID&#39;, &#39;Gene Symbol&#39;)]</code></pre>
<p>For my project I need to work only with cancer samples, so we explore <strong>pheno data</strong>, stored in the slot <code>pData</code> to identify them.</p>
<pre class="r"><code>phenodata &lt;- pData(x)

# From here https://stackoverflow.com/questions/51316238/find-substring-in-dataframe-in-r I found I nice way to identify the position of a string in the entire dataframe; 
# I am looking for the word tumor (or similar) that usually is not in the colnames
datatable(which(array(grepl(&quot;\\btumor\\b&quot;,as.matrix(phenodata)),dim(phenodata)),T))</code></pre>
<div id="htmlwidget-1" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-1">{"x":{"filter":"none","data":[[1,3,4,5,6,7,8,9,10,11,12,14,15,16,17,18,19,20,21,22,23,24,26,27,28,29,30,31,32,33,34,35,36,37,38,39,41,42,43,44,45,46,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,106,107,109,110,111,113,114,115,116,117,119,120,121,122,124,125,126,127,129,130,131,1,3,4,5,6,7,8,9,10,11,12,14,15,16,17,18,19,20,21,22,23,24,26,27,28,29,30,31,32,33,34,35,36,37,38,39,41,42,43,44,45,46,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,106,107,109,110,111,113,114,115,116,117,119,120,121,122,124,125,126,127,129,130,131],[10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,10,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39,39]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>row<\/th>\n      <th>col<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[0,1]}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code># The word is included in the columns 10 and 39; let&#39;s check them
datatable(phenodata[,c(10,39)])</code></pre>
<div id="htmlwidget-2" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-2">{"x":{"filter":"none","data":[["GSM1520738","GSM1520739","GSM1520740","GSM1520741","GSM1520742","GSM1520743","GSM1520744","GSM1520745","GSM1520746","GSM1520747","GSM1520748","GSM1520749","GSM1520750","GSM1520751","GSM1520752","GSM1520753","GSM1520754","GSM1520755","GSM1520756","GSM1520757","GSM1520758","GSM1520759","GSM1520760","GSM1520761","GSM1520762","GSM1520763","GSM1520764","GSM1520765","GSM1520766","GSM1520767","GSM1520768","GSM1520769","GSM1520770","GSM1520771","GSM1520772","GSM1520773","GSM1520774","GSM1520775","GSM1520776","GSM1520777","GSM1520778","GSM1520779","GSM1520780","GSM1520781","GSM1520782","GSM1520783","GSM1520784","GSM1520785","GSM1520786","GSM1520787","GSM1520788","GSM1520789","GSM1520790","GSM1520791","GSM1520792","GSM1520793","GSM1520794","GSM1520795","GSM1520796","GSM1520797","GSM1520798","GSM1520799","GSM1520800","GSM1520801","GSM1520802","GSM1520803","GSM1520804","GSM1520805","GSM1520806","GSM1520807","GSM1520808","GSM1520809","GSM1520810","GSM1520811","GSM1520812","GSM1520813","GSM1520814","GSM1520815","GSM1520816","GSM1520817","GSM1520818","GSM1520819","GSM1520820","GSM1520821","GSM1520822","GSM1520823","GSM1520824","GSM1520825","GSM1520826","GSM1520827","GSM1520828","GSM1520829","GSM1520830","GSM1520831","GSM1520832","GSM1520833","GSM1520834","GSM1520835","GSM1520836","GSM1520837","GSM1520838","GSM1520839","GSM1520840","GSM1520841","GSM1520842","GSM1520843","GSM1520844","GSM1520845","GSM1520846","GSM1520847","GSM1520848","GSM1520849","GSM1520850","GSM1520851","GSM1520852","GSM1520853","GSM1520854","GSM1520855","GSM1520856","GSM1520857","GSM1520858","GSM1520859","GSM1520860","GSM1520861","GSM1520862","GSM1520863","GSM1520864","GSM1520865","GSM1520866","GSM1520867","GSM1520868"],["tissue: pancreatic tumor","tissue: non-tumoral pancreatic tissue","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: non-tumoral pancreatic tissue","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: non-tumoral pancreatic tissue","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: non-tumoral pancreatic tissue","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: non-tumoral pancreatic tissue","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: non-tumoral pancreatic tissue","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: non-tumoral pancreatic tissue","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: non-tumoral pancreatic tissue","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: non-tumoral pancreatic tissue","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: non-tumoral pancreatic tissue","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: non-tumoral pancreatic tissue","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: non-tumoral pancreatic tissue","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: non-tumoral pancreatic tissue","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor"],["pancreatic tumor","non-tumoral pancreatic tissue","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","non-tumoral pancreatic tissue","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","non-tumoral pancreatic tissue","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","non-tumoral pancreatic tissue","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","non-tumoral pancreatic tissue","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","non-tumoral pancreatic tissue","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","non-tumoral pancreatic tissue","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","non-tumoral pancreatic tissue","pancreatic tumor","pancreatic tumor","non-tumoral pancreatic tissue","pancreatic tumor","pancreatic tumor","pancreatic tumor","non-tumoral pancreatic tissue","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","non-tumoral pancreatic tissue","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","non-tumoral pancreatic tissue","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","non-tumoral pancreatic tissue","pancreatic tumor","pancreatic tumor","pancreatic tumor"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>characteristics_ch1<\/th>\n      <th>tissue:ch1<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"order":[],"autoWidth":false,"orderClasses":false,"columnDefs":[{"orderable":false,"targets":0}]}},"evals":[],"jsHooks":[]}</script>

    </div>
  </article>

  <div id="disqus_thread"></div>
<script>
(function() {
var d = document, s = d.createElement('script');
s.src = 'https://pietrod.disqus.com/embed.js';
s.setAttribute('data-timestamp', +new Date());
(d.head || d.body).appendChild(s);
})();
</script>
<noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>

</main>

      <footer class="footer">
        <ul class="footer-links">
          <li>
            <a href="/index.xml" type="application/rss+xml" target="_blank">RSS feed</a>
          </li>
          <li>
            <a href="https://gohugo.io/" class="footer-links-kudos">Made with <img src="/images/hugo-logo.png" alt="Img link to Hugo website" width="22" height="22"></a>
          </li>
        </ul>
      </footer>

    </div>
    



<script src="//cdn.bootcss.com/highlight.js/9.11.0/highlight.min.js"></script>



<script src="//cdn.bootcss.com/highlight.js/9.11.0/languages/r.min.js"></script>
<script src="//cdn.bootcss.com/highlight.js/9.11.0/languages/yaml.min.js"></script>
<script>hljs.configure({languages: []}); hljs.initHighlightingOnLoad();</script>



    
<script src="/js/math-code.js"></script>
<script async src="//cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML"></script>


    
<script type="application/javascript">
var doNotTrack = false;
if (!doNotTrack) {
	(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
	(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
	m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
	})(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
	ga('create', 'UA-119674869-2', 'auto');
	
	ga('send', 'pageview');
}
</script>

  </body>
</html>

