<!DOCTYPE html>
<html lang="en-us">
  <head>
    <meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="generator" content="Hugo 0.55.6" />


<title>Gene expression - microarray data still useful - Pietro&#39;s website</title>
<meta property="og:title" content="Gene expression - microarray data still useful - Pietro&#39;s website">



  








<link href='//cdn.bootcss.com/highlight.js/9.11.0/styles/github.min.css' rel='stylesheet' type='text/css' />



<link rel="stylesheet" href="/css/fonts.css" media="all">
<link rel="stylesheet" href="/css/main.css" media="all">

<link rel="stylesheet" href="/css/mainoverwrite.css?v=1">



  </head>
  <body>
    <div class="wrapper">
      <header class="header">
        <nav class="nav">
  <a href="/" class="nav-logo">
    <img src="/images/pietro3.png"
         width="500"
         height="500"
         alt="Logo">
  </a>

  <ul class="nav-links">
    
    <li><a href="/about/">About</a></li>
    
    <li><a href="https://github.com/pietrod">GitHub</a></li>
    
    <li><a href="https://twitter.com/Pietro_Delfino">Twitter</a></li>
    
  </ul>
</nav>

      </header>


<main class="content" role="main">

  <article class="article">
    
    <span class="article-duration">9 min read</span>
    

    <h1 class="article-title">Gene expression - microarray data still useful</h1>

    
    <span class="article-date">2019/07/18</span>
    

    <div class="article-content">
      
<script src="/rmarkdown-libs/htmlwidgets/htmlwidgets.js"></script>
<script src="/rmarkdown-libs/jquery/jquery.min.js"></script>
<link href="/rmarkdown-libs/datatables-css/datatables-crosstalk.css" rel="stylesheet" />
<script src="/rmarkdown-libs/datatables-binding/datatables.js"></script>
<link href="/rmarkdown-libs/dt-core/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="/rmarkdown-libs/dt-core/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="/rmarkdown-libs/dt-core/js/jquery.dataTables.min.js"></script>
<link href="/rmarkdown-libs/crosstalk/css/crosstalk.css" rel="stylesheet" />
<script src="/rmarkdown-libs/crosstalk/js/crosstalk.min.js"></script>


<p>I work on pancreatic cancer <strong>-omics</strong>. Data mining on published data is a real source of gold today when someone wants to find confirmation of an hypothesis on a more general scale than his own dataset.</p>
<p>The most common -omics datasets out there are transcriptomics one; both RNA-Seq and good ol’ microarray. Regarding RNA-Seq, the well known TCGA and ICGC are my starting point for any explorative data mining analysis.</p>
<p>Regarding microarray, few studies with clear data have been published in the past, and someone has already collected the most interesting ones (see <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5674743/">Birnbaum et al. 2017</a>, <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5975421/">Zhao et al. 2018</a>).</p>
<p>For the scopes of my research I have also explored some of these datasets. The majority of theme are stored in the <a href="https://www.ncbi.nlm.nih.gov/geo/">GEO database</a>.</p>
<p>Here I want to present a basic workflow to retrieve a microarray expression dataset from GEO and to perform basic explorative analysis in <strong><em>R</em></strong>. As an example I take <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE62165">GSE62165</a> dataset.</p>
<p>To download and access data I use the <em>Bioconductor</em> <code>GEOquery</code> package (<a href="https://bioconductor.org/packages/release/bioc/html/GEOquery.html" class="uri">https://bioconductor.org/packages/release/bioc/html/GEOquery.html</a>).</p>
<pre class="r"><code># Load some libraries
library(tidyverse)
library(reshape2)
library(ggsci)
library(GEOquery)

# Downlaod ExpressionSet
x &lt;- getGEO(&#39;GSE62165&#39;, destdir = &#39;your/favourite/path&#39;, getGPL = T)</code></pre>
<p>We have downloaded the <em>GSE series matrix</em> to file and the <em>GPL .soft annotation</em>.
Now we load them back.</p>
<pre class="r"><code>x &lt;- getGEO(filename = &#39;your/favourite/path/GSE62165_series_matrix.txt.gz&#39;)</code></pre>
<p>Then we can start exploring the data.</p>
<pre class="r"><code># First we save the expression matrix in a different object
mat &lt;- exprs(x)

dim(mat)</code></pre>
<p>[1] 49386 131</p>
<table>
<caption><span id="tab:unnamed-chunk-5">Table 1: </span>Few lines of the expression matrix</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">GSM1520738</th>
<th align="right">GSM1520739</th>
<th align="right">GSM1520740</th>
<th align="right">GSM1520741</th>
<th align="right">GSM1520742</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>11715100_at</td>
<td align="right">3.412714</td>
<td align="right">3.736906</td>
<td align="right">3.300559</td>
<td align="right">3.223035</td>
<td align="right">3.073504</td>
</tr>
<tr class="even">
<td>11715101_s_at</td>
<td align="right">4.345207</td>
<td align="right">3.747892</td>
<td align="right">4.470069</td>
<td align="right">3.994475</td>
<td align="right">4.366361</td>
</tr>
<tr class="odd">
<td>11715102_x_at</td>
<td align="right">3.054823</td>
<td align="right">3.076922</td>
<td align="right">2.897363</td>
<td align="right">3.175010</td>
<td align="right">2.695467</td>
</tr>
<tr class="even">
<td>11715103_x_at</td>
<td align="right">4.510326</td>
<td align="right">4.156103</td>
<td align="right">3.456344</td>
<td align="right">4.064235</td>
<td align="right">4.227728</td>
</tr>
<tr class="odd">
<td>11715104_s_at</td>
<td align="right">3.229649</td>
<td align="right">3.367871</td>
<td align="right">3.030241</td>
<td align="right">3.249097</td>
<td align="right">2.973494</td>
</tr>
<tr class="even">
<td>11715105_at</td>
<td align="right">4.554059</td>
<td align="right">3.486847</td>
<td align="right">3.294781</td>
<td align="right">3.137451</td>
<td align="right">2.961364</td>
</tr>
<tr class="odd">
<td>11715106_x_at</td>
<td align="right">3.103925</td>
<td align="right">3.377799</td>
<td align="right">3.467268</td>
<td align="right">3.585230</td>
<td align="right">3.940851</td>
</tr>
<tr class="even">
<td>11715107_s_at</td>
<td align="right">3.082911</td>
<td align="right">4.107503</td>
<td align="right">3.012626</td>
<td align="right">3.033756</td>
<td align="right">3.222768</td>
</tr>
</tbody>
</table>
<p>Then we explore the distrbution of samples’ expression values with a boxplot.</p>
<pre class="r"><code>boxplot(mat, outline=F, main=&#39;GSE62165&#39;)</code></pre>
<p><img src="/post/2019-07-17-microarray-still-useful_files/figure-html/unnamed-chunk-6-1.svg" width="672" /></p>
<p>What about annotation data? They are stored in <code>fData</code> slot. We save them in a unique object and look for the <strong>Gene Symbol</strong> column, that we will use to replace row names (probes id).</p>
<pre class="r"><code>ann &lt;- fData(x)

colnames(ann)</code></pre>
<p>[1] “ID” “GeneChip Array”<br />
[3] “Species Scientific Name” “Annotation Date”<br />
[5] “Sequence Type” “Sequence Source”<br />
[7] “Transcript ID(Array Design)” “Target Description”<br />
[9] “Representative Public ID” “Archival UniGene Cluster”<br />
[11] “UniGene ID” “Genome Version”<br />
[13] “Alignments” “Gene Title”<br />
[15] “Gene Symbol” “Chromosomal Location”<br />
[17] “GB_LIST” “SPOT_ID”<br />
[19] “Unigene Cluster Type” “Ensembl”<br />
[21] “Entrez Gene” “SwissProt”<br />
[23] “EC” “OMIM”<br />
[25] “RefSeq Protein ID” “RefSeq Transcript ID”<br />
[27] “FlyBase” “AGI”<br />
[29] “WormBase” “MGI Name”<br />
[31] “RGD Name” “SGD accession number”<br />
[33] “Gene Ontology Biological Process” “Gene Ontology Cellular Component”
[35] “Gene Ontology Molecular Function” “Pathway”<br />
[37] “InterPro” “Trans Membrane”<br />
[39] “QTL” “Annotation Description”<br />
[41] “Annotation Transcript Cluster” “Transcript Assignments”<br />
[43] “Annotation Notes”</p>
<pre class="r"><code># Subset to columns we&#39;ll use
ann &lt;- ann[,c(&#39;ID&#39;, &#39;Gene Symbol&#39;)]</code></pre>
<p>For my project I need to work only with cancer samples, so we explore <strong>pheno data</strong>, stored in the slot <code>pData</code> to identify them.</p>
<pre class="r"><code>phenodata &lt;- pData(x)

# From here https://stackoverflow.com/questions/51316238/find-substring-in-dataframe-in-r I found I nice way to identify the position of a string in the entire dataframe; 
# I am looking for the word tumor (or similar) that usually is not in the colnames
unique(which(array(grepl(&quot;\\btumor\\b&quot;,as.matrix(phenodata)),dim(phenodata)),T)[,2])</code></pre>
<pre><code>## [1] 10 39</code></pre>
<p>The word is included in the columns 10 and 39; let’s check them.</p>
<div id="htmlwidget-1" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-1">{"x":{"filter":"none","data":[["GSM1520738","GSM1520739","GSM1520740","GSM1520741","GSM1520742","GSM1520743","GSM1520744","GSM1520745","GSM1520746","GSM1520747","GSM1520748","GSM1520749","GSM1520750","GSM1520751","GSM1520752","GSM1520753","GSM1520754","GSM1520755","GSM1520756","GSM1520757","GSM1520758","GSM1520759","GSM1520760","GSM1520761","GSM1520762","GSM1520763","GSM1520764","GSM1520765","GSM1520766","GSM1520767","GSM1520768","GSM1520769","GSM1520770","GSM1520771","GSM1520772","GSM1520773","GSM1520774","GSM1520775","GSM1520776","GSM1520777","GSM1520778","GSM1520779","GSM1520780","GSM1520781","GSM1520782","GSM1520783","GSM1520784","GSM1520785","GSM1520786","GSM1520787","GSM1520788","GSM1520789","GSM1520790","GSM1520791","GSM1520792","GSM1520793","GSM1520794","GSM1520795","GSM1520796","GSM1520797","GSM1520798","GSM1520799","GSM1520800","GSM1520801","GSM1520802","GSM1520803","GSM1520804","GSM1520805","GSM1520806","GSM1520807","GSM1520808","GSM1520809","GSM1520810","GSM1520811","GSM1520812","GSM1520813","GSM1520814","GSM1520815","GSM1520816","GSM1520817","GSM1520818","GSM1520819","GSM1520820","GSM1520821","GSM1520822","GSM1520823","GSM1520824","GSM1520825","GSM1520826","GSM1520827","GSM1520828","GSM1520829","GSM1520830","GSM1520831","GSM1520832","GSM1520833","GSM1520834","GSM1520835","GSM1520836","GSM1520837","GSM1520838","GSM1520839","GSM1520840","GSM1520841","GSM1520842","GSM1520843","GSM1520844","GSM1520845","GSM1520846","GSM1520847","GSM1520848","GSM1520849","GSM1520850","GSM1520851","GSM1520852","GSM1520853","GSM1520854","GSM1520855","GSM1520856","GSM1520857","GSM1520858","GSM1520859","GSM1520860","GSM1520861","GSM1520862","GSM1520863","GSM1520864","GSM1520865","GSM1520866","GSM1520867","GSM1520868"],["tissue: pancreatic tumor","tissue: non-tumoral pancreatic tissue","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: non-tumoral pancreatic tissue","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: non-tumoral pancreatic tissue","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: non-tumoral pancreatic tissue","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: non-tumoral pancreatic tissue","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: non-tumoral pancreatic tissue","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: non-tumoral pancreatic tissue","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: non-tumoral pancreatic tissue","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: non-tumoral pancreatic tissue","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: non-tumoral pancreatic tissue","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: non-tumoral pancreatic tissue","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: non-tumoral pancreatic tissue","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: non-tumoral pancreatic tissue","tissue: pancreatic tumor","tissue: pancreatic tumor","tissue: pancreatic tumor"],["pancreatic tumor","non-tumoral pancreatic tissue","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","non-tumoral pancreatic tissue","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","non-tumoral pancreatic tissue","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","non-tumoral pancreatic tissue","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","non-tumoral pancreatic tissue","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","non-tumoral pancreatic tissue","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","non-tumoral pancreatic tissue","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","non-tumoral pancreatic tissue","pancreatic tumor","pancreatic tumor","non-tumoral pancreatic tissue","pancreatic tumor","pancreatic tumor","pancreatic tumor","non-tumoral pancreatic tissue","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","non-tumoral pancreatic tissue","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","non-tumoral pancreatic tissue","pancreatic tumor","pancreatic tumor","pancreatic tumor","pancreatic tumor","non-tumoral pancreatic tissue","pancreatic tumor","pancreatic tumor","pancreatic tumor"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>characteristics_ch1<\/th>\n      <th>tissue:ch1<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"order":[],"autoWidth":false,"orderClasses":false,"columnDefs":[{"orderable":false,"targets":0}]}},"evals":[],"jsHooks":[]}</script>
<p>Subset to the columns we will need. In particular column <code>tissue:ch1</code> that has a more clear nomenclature.</p>
<pre class="r"><code># We also rename the column
phenodata &lt;- data.frame(tissue = phenodata[,39])</code></pre>
<p>We use the info contained in this column to subset the matrix to only the cancer samples. We remember that original matrix had 131 samples (columns).</p>
<pre class="r"><code>mat &lt;- mat[, !grepl(&#39;non-tumoral&#39;, phenodata$tissue)]

dim(mat)</code></pre>
<pre><code>## [1] 49386   118</code></pre>
<p>We are left with 118 samples.</p>
<p>Since my final aim is to apply some particular functions that need one row per gene, we have to face the neverending problem of having multiple probes matching to the same gene. We saw that in total we have more than 40K probe IDs. So we surely have genes with more than one probe matching.</p>
<p>Here I apply the average over multiple probes. I know that this is an approximation, but here we really need to have a single gene for each patients (also I take with very high consideration the words from Gordon Smyth <a href="https://support.bioconductor.org/p/44161/" class="uri">https://support.bioconductor.org/p/44161/</a>)</p>
<p>I know that there is the <code>avereps</code> function from <code>limma</code> package, but I like to write my own scripts, so I wrote a script with the <strong>tidyverse</strong> style that makes use of <code>group_by</code> and <code>summarize</code> functions. We use the annotation df to find out which genes have multiple probes.</p>
<pre class="r"><code># Save to new object
mat_mean &lt;- mat %&gt;% 
    as.data.frame() %&gt;% 
    rownames_to_column(&#39;ID&#39;) %&gt;%  
    left_join(ann, by=&#39;ID&#39;) %&gt;% 
    dplyr::select(-ID) %&gt;% 
    dplyr::rename(SYMBOL = &#39;Gene Symbol&#39;) %&gt;%
    # remove NAs
    dplyr::filter(!is.na(SYMBOL)) %&gt;%
    # remove empty values
    dplyr::filter(!SYMBOL == &#39;&#39;) %&gt;%
    # find the groups by genes
    group_by(SYMBOL) %&gt;% 
    # apply mean by these groups 
    summarize_all(.funs = mean, na.rm=T) %&gt;%  
    # go back to matrix format
    column_to_rownames(&#39;SYMBOL&#39;) %&gt;% 
    as.matrix()</code></pre>
<table>
<caption><span id="tab:unnamed-chunk-13">Table 2: </span>Few lines of the expression matrix with averaged probes</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">GSM1520738</th>
<th align="right">GSM1520740</th>
<th align="right">GSM1520741</th>
<th align="right">GSM1520742</th>
<th align="right">GSM1520743</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>A1BG</td>
<td align="right">4.051629</td>
<td align="right">3.782756</td>
<td align="right">3.877025</td>
<td align="right">3.782277</td>
<td align="right">3.578368</td>
</tr>
<tr class="even">
<td>A1CF</td>
<td align="right">4.193239</td>
<td align="right">6.280513</td>
<td align="right">4.261783</td>
<td align="right">5.393971</td>
<td align="right">8.266222</td>
</tr>
<tr class="odd">
<td>A2BP1</td>
<td align="right">2.934138</td>
<td align="right">2.987857</td>
<td align="right">2.800693</td>
<td align="right">2.818495</td>
<td align="right">2.879545</td>
</tr>
<tr class="even">
<td>A2LD1</td>
<td align="right">4.770593</td>
<td align="right">4.810682</td>
<td align="right">4.257089</td>
<td align="right">4.081143</td>
<td align="right">4.947301</td>
</tr>
<tr class="odd">
<td>A2M</td>
<td align="right">11.657892</td>
<td align="right">12.022625</td>
<td align="right">11.120334</td>
<td align="right">11.950187</td>
<td align="right">11.757543</td>
</tr>
<tr class="even">
<td>A2ML1</td>
<td align="right">2.501171</td>
<td align="right">2.633293</td>
<td align="right">7.301564</td>
<td align="right">3.749501</td>
<td align="right">2.653799</td>
</tr>
<tr class="odd">
<td>A3GALT2</td>
<td align="right">2.868238</td>
<td align="right">2.683267</td>
<td align="right">2.773406</td>
<td align="right">2.974176</td>
<td align="right">2.721883</td>
</tr>
</tbody>
</table>
<p>Now we have a sort of gene expression matrix ready to be used for cool things.
Before, I would like to show another <code>tidyverse</code> method for obtaining histograms for each sample of the expression matrix. Sometimes it is useful, besides boxplot, to spot strange samples.</p>
<pre class="r"><code># Since we are dealing with a lot of samples, we&#39;ll plot a subset
mat_mean[,1:16] %&gt;% 
    as.data.frame() %&gt;%
    rownames_to_column(&#39;ID&#39;) %&gt;% 
    melt() %&gt;%
    
    ggplot(aes(value)) +
    geom_histogram() +
    facet_wrap(~variable)</code></pre>
<p><img src="/post/2019-07-17-microarray-still-useful_files/figure-html/unnamed-chunk-14-1.svg" width="672" /></p>
<p>To make results more comparable across datasets, something very useful especially when doing deep data mining (in total I collected more than 1300 pancreatic cancer samples from 16 experiments) is to obtain <em>Z-scores</em> from the expression matrix.</p>
<p>From the <code>phenoData</code> slot we appreciate that the expression data we downloaded were already normalized and <code>log2</code> transformed.</p>
<p>The standardization procedure (<em>Z-scores</em>) is usually performed on already normalized datasets. So we can apply it straight to the <code>mat_mean</code> object. Before we define a function.</p>
<pre class="r"><code>zscore &lt;- function(x) {
    z &lt;- (x - mean(x, na.rm=T)) / sd(x, na.rm=T)
    return(z)
}</code></pre>
<p>Then we obtain the <em>Z-score</em> matrix.</p>
<pre class="r"><code>mat_zscore_global &lt;- zscore(mat_mean)</code></pre>
<p>From the above function you can see that the standardization is performed considering overall mean and overall standard deviation, that is, calculated all across the dataset. If you search online and on forums you will find different ways of obtaining <em>Z-scores</em> from gene expression studies (just a quick <em>Google</em> search will reveal quite a few discussion especially on <strong>Biostars</strong>). The main point here is the difference between a <em>Z-score</em> based on overall (population) statistics and one based on by row (that is, by gene) statistics. In brief, is it more correct to subtract and to divide a value by overall or by gene mean and SD? I have not found anything out there in literature about, only this <a href="https://stats.stackexchange.com/questions/201961/do-i-apply-normalization-per-entire-dataset-per-input-vector-or-per-feature">thread</a>. If you know something about, please comment!</p>
<p>If we wanted to obtain <em>Z-scores</em> by row (by gene), then the <code>scale</code> function would come to help, but coupled with a transposition, because by default it operates by column. So, something like</p>
<pre class="r"><code>mat_zscore_gene &lt;- t(scale(t(mat_mean)))</code></pre>
<p>In general, it would be useful that everytime someone reports gene expression data in <em>Z-scores</em> terms, to explicitly state how the standardization was performed. I am not myself a statistician, so I cannot say concretely what the difference would be, but to be extremely reproducible, something that every scientist should aim to, every method, even simple, should be described clearly.</p>
<pre class="r"><code>par(mfrow=c(1,2))

boxplot(mat_zscore_global, outline=F, main=&#39;Z-scores global&#39;)
boxplot(mat_zscore_gene, outline=F, main=&#39;Z-scores gene&#39;)</code></pre>
<p><img src="/post/2019-07-17-microarray-still-useful_files/figure-html/unnamed-chunk-18-1.svg" width="864" /></p>
<p>The two methods yield different results. It always amazes me how even simple things like that are not reported clearly in papers and then lead to different results when downstream analysis are performed. I will dig more into this.</p>
<p>When I say that they give different results I mean this:</p>
<pre class="r"><code>mat_zscore_global[c(&#39;ACTB&#39;, &#39;KRAS&#39;, &#39;IGFBP4&#39;),] %&gt;% 
    as.data.frame() %&gt;% 
    rownames_to_column(&#39;SYMBOL&#39;) %&gt;% 
    mutate(`Z-score` = &#39;Global&#39;) %&gt;% 
    bind_rows( mat_zscore_gene[c(&#39;ACTB&#39;, &#39;KRAS&#39;, &#39;IGFBP4&#39;),] %&gt;% 
    as.data.frame() %&gt;% 
    rownames_to_column(&#39;SYMBOL&#39;) %&gt;% 
    mutate(`Z-score` = &#39;Gene&#39;) ) %&gt;% 
    melt() %&gt;% 
    
    ggplot(aes(SYMBOL, value, fill=`Z-score`)) +
    geom_boxplot(outlier.shape = NA) +
    
    stat_compare_means(aes(group=`Z-score`), label = &quot;p.format&quot;, method = &#39;t.test&#39;) +
    
    theme_bw() +
    theme(axis.title.x = element_blank(),
          axis.text.x = element_text(size=14, color=&#39;black&#39;),
          axis.text.y = element_text(size=14, color=&#39;black&#39;)) +
    xlab(&#39;&#39;) +
    ylab(&#39;&#39;) +
    scale_fill_npg()</code></pre>
<p><img src="/post/2019-07-17-microarray-still-useful_files/figure-html/unnamed-chunk-19-1.svg" width="768" /></p>

    </div>
  </article>

  <div id="disqus_thread"></div>
<script>
(function() {
var d = document, s = d.createElement('script');
s.src = 'https://pietrod.disqus.com/embed.js';
s.setAttribute('data-timestamp', +new Date());
(d.head || d.body).appendChild(s);
})();
</script>
<noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>

</main>

      <footer class="footer">
        <ul class="footer-links">
          <li>
            <a href="/index.xml" type="application/rss+xml" target="_blank">RSS feed</a>
          </li>
          <li>
            <a href="https://gohugo.io/" class="footer-links-kudos">Made with <img src="/images/hugo-logo.png" alt="Img link to Hugo website" width="22" height="22"></a>
          </li>
        </ul>
      </footer>

    </div>
    



<script src="//cdn.bootcss.com/highlight.js/9.11.0/highlight.min.js"></script>



<script src="//cdn.bootcss.com/highlight.js/9.11.0/languages/r.min.js"></script>
<script src="//cdn.bootcss.com/highlight.js/9.11.0/languages/yaml.min.js"></script>
<script>hljs.configure({languages: []}); hljs.initHighlightingOnLoad();</script>



    
<script src="/js/math-code.js"></script>
<script async src="//cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML"></script>


    
<script type="application/javascript">
var doNotTrack = false;
if (!doNotTrack) {
	(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
	(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
	m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
	})(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
	ga('create', 'UA-119674869-2', 'auto');
	
	ga('send', 'pageview');
}
</script>

  </body>
</html>

