<!DOCTYPE html>
<html lang="en-us">
  <head>
    <meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<meta name="generator" content="Hugo 0.48" />


<title>Principal component analysis with any kind of data in R - Pietro&#39;s website</title>
<meta property="og:title" content="Principal component analysis with any kind of data in R - Pietro&#39;s website">



  








<link href='//cdn.bootcss.com/highlight.js/9.11.0/styles/github.min.css' rel='stylesheet' type='text/css' />



<link rel="stylesheet" href="/css/fonts.css" media="all">
<link rel="stylesheet" href="/css/main.css" media="all">



  </head>
  <body>
    <div class="wrapper">
      <header class="header">
        <nav class="nav">
  <a href="/" class="nav-logo">
    <img src="/images/logo.png"
         width="50"
         height="50"
         alt="Logo">
  </a>

  <ul class="nav-links">
    
    <li><a href="/about/">About</a></li>
    
    <li><a href="/cv/">CV</a></li>
    
    <li><a href="https://github.com/pietrod">GitHub</a></li>
    
    <li><a href="https://twitter.com/Pietro_Delfino">Twitter</a></li>
    
  </ul>
</nav>

      </header>


<main class="content" role="main">

  <article class="article">
    
    <span class="article-duration">2 min read</span>
    

    <h1 class="article-title">Principal component analysis with any kind of data in R</h1>

    
    <span class="article-date">2018/09/05</span>
    

    <div class="article-content">
      <div id="principal-component-analysis-with-various-data---tidyverse-way" class="section level1">
<h1>Principal Component Analysis with various data - tidyverse way</h1>
<p>In my daily work I am often asked to perform Principal Component Analysis of various type of <em>-omics</em> data, for example data coming from RNA-Seq.</p>
<p>With this data usually the rows correspond to genes (transcripts, proteins, etc..) while columns are individuals or samples.</p>
<p>For example</p>
<pre class="r"><code>x</code></pre>
<pre><code>## # A tibble: 846 x 37
##      Row Index `Peak Name` `m/z` `Ret. Time` Group Use   `Coltura organo…
##    &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt;       &lt;chr&gt; &lt;chr&gt;       &lt;chr&gt; &lt;chr&gt;            &lt;dbl&gt;
##  1    NA    NA &lt;NA&gt;        &lt;NA&gt;  &lt;NA&gt;        &lt;NA&gt;  Group               5 
##  2     1     1 sp|P19137|… N/A   N/A         Lami… TRUE            92493.
##  3     2     2 sp|P02469|… N/A   N/A         sp|P… TRUE           101139.
##  4     3     3 sp|Q80X90|… N/A   N/A         Fila… TRUE           104487.
##  5     4     4 sp|P02468|… N/A   N/A         sp|P… TRUE           133946.
##  6     5     5 sp|P60710|… N/A   N/A         sp|P… TRUE           479363.
##  7     6     6 sp|P52480|… N/A   N/A         sp|P… TRUE           372423.
##  8     7     7 sp|P16858|… N/A   N/A         sp|P… TRUE           646268.
##  9     8     8 sp|Q8BTM8|… N/A   N/A         sp|Q… TRUE           111774.
## 10     9     9 sp|Q8VHX6-… N/A   N/A         sp|Q… TRUE             5144.
## # ... with 836 more rows, and 29 more variables: `Coltura organotipica
## #   TOPO_KC_DMEM_1ug_31072018_SWATH2` &lt;dbl&gt;, `Coltura organotipica
## #   TOPO_KC_DMEM_1ug_31072018_SWATH3` &lt;dbl&gt;, `Coltura organotipica
## #   TOPO_KC_HFD_1ug_31072018_SWATH2` &lt;dbl&gt;, `Coltura organotipica
## #   TOPO_KC_HFD_1ug_31072018_SWATH1` &lt;dbl&gt;, `Coltura organotipica
## #   TOPO_KC_HFD_1ug_31072018_SWATH3` &lt;dbl&gt;, `Coltura organotipica
## #   TOPO_KC_LFD_1ug_31072018_SWATH1` &lt;dbl&gt;, `Coltura organotipica
## #   TOPO_KC_LFD_1ug_31072018_SWATH2` &lt;dbl&gt;, `Coltura organotipica
## #   TOPO_KC_LFD_1ug_31072018_SWATH3` &lt;dbl&gt;, `Coltura organotipica
## #   TOPO_KC_MW_1ug_30072018_SWATH1` &lt;dbl&gt;, `Coltura organotipica
## #   TOPO_KC_MW_1ug_30072018_SWATH2` &lt;dbl&gt;, `Coltura organotipica
## #   TOPO_KC_MW_1ug_30072018_SWATH3` &lt;dbl&gt;, `Coltura organotipica
## #   TOPO_KC_ObOb_2ug_02082018_SWATH1` &lt;dbl&gt;, `Coltura organotipica
## #   TOPO_KC_ObOb_2ug_02082018_SWATH2` &lt;dbl&gt;, `Coltura organotipica
## #   TOPO_KC_ObOb_2ug_02082018_SWATH3` &lt;dbl&gt;, `Coltura organotipica
## #   TOPO_KPC_DMEM_2ug_02082018_SWATH1` &lt;dbl&gt;, `Coltura organotipica
## #   TOPO_KPC_DMEM_2ug_02082018_SWATH2` &lt;dbl&gt;, `Coltura organotipica
## #   TOPO_KPC_DMEM_2ug_02082018_SWATH3` &lt;dbl&gt;, `Coltura organotipica
## #   TOPO_KPC_HFD_1ug_01082018_SWATH1` &lt;dbl&gt;, `Coltura organotipica
## #   TOPO_KPC_HFD_1ug_01082018_SWATH2` &lt;dbl&gt;, `Coltura organotipica
## #   TOPO_KPC_HFD_1ug_01082018_SWATH3` &lt;dbl&gt;, `Coltura organotipica
## #   TOPO_KPC_LFD_2ug_02082018_SWATH1` &lt;dbl&gt;, `Coltura organotipica
## #   TOPO_KPC_LFD_2ug_02082018_SWATH2` &lt;dbl&gt;, `Coltura organotipica
## #   TOPO_KPC_LFD_2ug_02082018_SWATH3` &lt;dbl&gt;, `Coltura organotipica
## #   TOPO_KPC_MW_1ug_30072018_SWATH1` &lt;dbl&gt;, `Coltura organotipica
## #   TOPO_KPC_MW_1ug_30072018_SWATH2` &lt;dbl&gt;, `Coltura organotipica
## #   TOPO_KPC_MW_1ug_30072018_SWATH3` &lt;dbl&gt;, `Coltura organotipica
## #   TOPO_KPC_ObOb_2ug_02082018_SWATH1` &lt;dbl&gt;, `Coltura organotipica
## #   TOPO_KPC_ObOb_2ug_02082018_SWATH2` &lt;dbl&gt;, `Coltura organotipica
## #   TOPO_KPC_ObOb_2ug_02082018_SWATH3` &lt;dbl&gt;</code></pre>
</div>

    </div>
  </article>

  


</main>

      <footer class="footer">
        <ul class="footer-links">
          <li>
            <a href="/index.xml" type="application/rss+xml" target="_blank">RSS feed</a>
          </li>
          <li>
            <a href="https://gohugo.io/" class="footer-links-kudos">Made with <img src="/images/hugo-logo.png" alt="Img link to Hugo website" width="22" height="22"></a>
          </li>
        </ul>
      </footer>

    </div>
    



<script src="//cdn.bootcss.com/highlight.js/9.11.0/highlight.min.js"></script>



<script src="//cdn.bootcss.com/highlight.js/9.11.0/languages/r.min.js"></script>
<script src="//cdn.bootcss.com/highlight.js/9.11.0/languages/yaml.min.js"></script>
<script>hljs.configure({languages: []}); hljs.initHighlightingOnLoad();</script>



    

    
  </body>
</html>

